src/core/mempool.rs

```diff
@@ -956,11 +1114,124 @@ impl MemPoolDB {
         }
     }
 
+    /// Remove all txids at the given height from the bloom counter.
+    /// Used to clear out txids that are now outside the bloom counter's depth.
+    fn prune_bloom_counter(tx: &mut MemPoolTx, target_height: u64) -> Result<(), MemPoolRejection> {
+        let sql = "SELECT txid FROM mempool WHERE height = ?1 AND NOT EXISTS (SELECT 1 FROM removed_txids WHERE txid = mempool.txid)";
+        let args: &[&dyn ToSql] = &[&u64_to_sql(target_height)?];
+        let txids: Vec<Txid> = query_rows(tx, sql, args)?;
+        let num_txs = txids.len();
+
+        debug!("Prune bloom counter from height {}", target_height);
+
+        // keep borrow-checker happy
+        let bloom_counter = tx.take_bloom_state();
+        for txid in txids.into_iter() {
+            bloom_counter.remove_raw(&mut tx.tx, &txid.0)?;
+
+            let sql = "INSERT OR REPLACE INTO removed_txids (txid) VALUES (?1)";
+            let args: &[&dyn ToSql] = &[&txid];
+            tx.execute(sql, args).map_err(db_error::SqliteError)?;
```

Comment:
If these early exit, it seems like the bloom state could be corrupted (i.e., not replaced). I think the easiest way to make this "safe" would be to add a `using!` macro (like in `clarity_vm/clarity.rs`).

Reply:
yup, will replace with safer code